yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/call/docs/DocScanService.java

```diff
@@ -0,0 +1,277 @@
+ * Doc Scan API's.
+ *
+ * @since 2.7
+ */
+public final class DocScanService {
+
+    public static DocScanService newInstance() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+        return new DocScanService(new UnsignedPathFactory(), objectMapper);
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(DocScanService.class);
+
+    private final UnsignedPathFactory unsignedPathFactory;
+    private final ObjectMapper objectMapper;
+
+    private final String apiUrl;
+
+    private DocScanService(UnsignedPathFactory pathFactory, ObjectMapper objectMapper) {
+        this.unsignedPathFactory = pathFactory;
+        this.objectMapper = objectMapper;
+
+        apiUrl = System.getProperty(PROPERTY_YOTI_DOCS_URL, DEFAULT_YOTI_DOCS_URL);
+    }
+
+    /**
+     * Performs a POST request using the supplied session specification to create
+     * a session, returning information about the newly created session if successful
+     *
+     * @param appId the app ID
+     * @param keyPair the {@code KeyPair}
+     * @param sessionSpec the {@code SessionSpec}
+     * @return the session creation result
+     * @throws DocScanException if there was an error whilst performing
+     *                          the request to Doc Scan systems
+     */
+    public CreateSessionResult createSession(String appId, KeyPair keyPair, SessionSpec sessionSpec) throws DocScanException {
+        notNullOrEmpty(appId, "Application id");
+        notNull(keyPair, "Application key Pair");
+        notNull(sessionSpec, "sessionSpec");
+
+        String path = unsignedPathFactory.createNewYotiDocsSessionPath(appId);
+        LOG.info("Creating session at '{}'", path);
+
+        try {
+            byte[] payload = objectMapper.writeValueAsBytes(sessionSpec);
+
+            SignedRequest signedRequest = getSignedRequestBuilder()
+                    .withKeyPair(keyPair)
+                    .withBaseUrl(apiUrl)
+                    .withEndpoint(path)
+                    .withHttpMethod(HTTP_POST)
+                    .withPayload(payload)
+                    .withHeader(CONTENT_TYPE, CONTENT_TYPE_JSON)
+                    .build();
+
+            return signedRequest.execute(SimpleCreateSessionResult.class);
+        } catch (GeneralSecurityException ex) {
+            throw new DocScanException("Error signing the request: " + ex.getMessage(), ex);
+        } catch (ResourceException ex) {
+            throw new DocScanException("Error posting the request: " + ex.getMessage(), ex);
+        } catch (IOException | URISyntaxException ex) {
+            throw new DocScanException("Error building the request: " + ex.getMessage(), ex);
+        } catch (Exception ex) {
+            throw new DocScanException("Error creating the session: " + ex.getMessage(), ex);
+        }
```

Comment:
I'm assuming we know conclusively that these exceptions can only be thrown in the respective sign/post/build/create parts of this? Will just make debugging a fair bit harder if for any reason these exceptions can be thrown from other parts of this process if we're appending `Error <x>ing the request/session`

Reply:
Ok, if we're confident that these exceptions won't be thrown at other stages then this is fine